src/store/store.js

```diff
@@ -355,15 +355,20 @@ var Store = (Ember.Service || Ember.Object).extend({
 			recordRequestCache.savePendingRequest(typeKey, query, promise);
 		}
 
-		return PromiseArray.create({
-			promise: promise.then((payload) => {
-				var records = payload.meta.matchedRecords;
-				this.pushPayload(payload);
-
-				return records.map((record) => {
-					return this.getRecord(record.type, record.id);
-				});
-			})
+		return promise.then((payload) => {
+			return {
```

Comment:
This would be a pretty big breaking change to the API. `PromiseArray` extends `ArrayProxy` which extends `Ember.Object`. If you want, you could just set a `meta` property on the array (or something similar). This would allow the return value of a query to jive well with templates. Up to you guys though, I'm not sure what your use cases are.